src/operand/reference.ts

```diff
@@ -137,15 +137,22 @@ export class Reference extends Operand {
   /**
    * {@link Evaluable.simplify}
    */
-  simplify(ctx: Context, ignoreKeys: string[]): Result | Evaluable {
+  simplify(
+    ctx: Context,
+    strictKeys?: string[],
+    optionalKeys?: string[]
+  ): Result | Evaluable {
     const keys = extractKeys(ctx, this.key)
 
     if (!keys) {
       return this
     }
 
     const key = keys[0].replace(/\[.+$/, '')
-    if (ctx[key] !== undefined || ignoreKeys.includes(key)) {
+    const shouldEvaluate =
+      (strictKeys && strictKeys.includes(key)) ||
+      (optionalKeys && !optionalKeys.includes(key))
+    if (ctx[key] !== undefined || shouldEvaluate) {
       return this.evaluate(ctx)
     }
```

Comment:
A small improvement:
- we don't need to call `this.evaluate` if `shouldEvaluate` is false. We know that it will return undefined. There is the case of complex keys, but that is not being handled here anyway.
- We don't need to check if the key is included when we know that the value is defined.

```suggestion
    if (ctx[key] !== undefined || shouldEvaluate) {
      return this.evaluate(ctx)
    } else if (
      (strictKeys && strictKeys.includes(key)) ||
      (optionalKeys && !optionalKeys.includes(key))
    ) {
      return undefined
    }
```

Reply:
Makes sense. Done in 5146b749ff24c2b6ac3efc2e538e678f39fbefcd